src/libutil/file-system.cc

```diff
@@ -29,7 +29,7 @@
 
 namespace nix {
 
-namespace fs { using namespace std::filesystem; }
+namespace fs = std::filesystem;
```

Comment:
What's going on here? There are very specific reasons for doing this I don't want to change just to make Doxygen happy.

Reply:
I wanted to make sure to never clone in in a header, precisely so it wouldn't show up in doxygen or downstream code.

Really what I wanted is to have `::nix::fs` and `::fs`, and C++ will simply check both in order resolve `fs::foo`, but that doesn't seem to be an option. I am open to other suggestions that move us towards using namespaces more (I don't just want to remove the `fs` from `symlink_exists`).

FWIW `symlink_exists` is also named to match `std::filesystem` conventions (like `status` vs `symlink_status`). I wasn't going to literally define it in `std::filesystem` of course, but I sort of think of it as "morally" a polyfill for a missing shorthand in that API.

Reply:
I think it's confusing to have a `nix::fs` that is almost but not quite `std::filesystem`. The doxygen warning is a symptom of that confusion (is `nix::fs::path` the same as `std::filesystem::path`?).

But I've removed the `nix::fs` changes from this PR.